### ISSUE TITLE
No text login screen with older video card hardware

### ISSUE THREAD

**Issue by rachid-a** [sddm/sddm#1958]
There is issue on sddm version 0.21.0_p20240302 and KDE plasma 6.1.1 (also older versions) with older video card Intel i915.
The issue is that there is no text on the login page and on the lock screen. Only pictures are showing.

In the journalctl log i see the following message:

sddm-greeter-qt6[1050]: Failed to build graphics pipeline state
sddm-greeter-qt6[1050]: Failed to link shader program: error: Exceeded max nr indirect texture lookups (5/4)

**Comment by Vogtinator** [comment]
If even SDDM doesn't start, Plasma likely won't work either. You can report this to Qt upstream to fall back to software rendering automatically or just force `LIBGL_ALWAYS_SOFTWARE=1` on your system.

**Comment by thesamesam** [comment]
The user's reported it to Qt at https://bugreports.qt.io/browse/QTBUG-126706 now. Thanks!

**Comment by rachid-a** [comment]
@Vogtinator KDE has fixed this on their part in Kwin with issues without Qt upstream change

From Qt they responded with:  "Not sure what Qt can do here. What do those shaders contain that makes the compiler choke?"
How can we provide the correct information to resolve this issue. Only part is sddm-greeter-qt6

**Comment by Vogtinator** [comment]
SDDM can't do anything here, it can't tell in advance what the requirements of the used theme are. It's also not possible to switch to software rendering after a QML rendering failure AFAIK.

It really needs to be done in Qt.

**Comment by vectorsigma** [comment]
I had this same problem with an old intel i915 laptop.  It's ugly, but I forced it to use software opengl rendering, and everything worked peachy after that.

Edit `/etc/security/pam_env.conf`

At the bottom of that file, add the line:
`QT_QUICK_BACKEND DEFAUILT="software"`

Save and close, `systemctl restart sddm` and it should come back with visible text.  Verified on Fedora 40's latest versions of SDDM and LXQt at the time of this writing.

Got the idea from [this SDDM bug](https://github.com/sddm/sddm/issues/575), HTT to @tachometer
